Error writing empty nested arrays

I was getting assertion errors while writing some data to disk and have reduced it to this minimal repro:

```
julia> Avro.write([Bool[]])
ERROR: AssertionError: nb == pos - startpos
Stacktrace:
 [1] writevalue(B::Avro.Binary, A::Avro.ArrayType, x::Vector{Vector{Bool}}, buf::Vector{UInt8}, pos::Int64, len::Int64, opts::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Avro ~/.julia/packages/Avro/JEoRa/src/types/arrays.jl:44
 [2] write(obj::Vector{Vector{Bool}}; schema::Avro.ArrayType, jsonencoding::Bool, kw::Base.Iterators.Pairs{Union{}, Union{}, Tuple{}, NamedTuple{(), Tuple{}}})
   @ Avro ~/.julia/packages/Avro/JEoRa/src/types/binary.jl:26
 [3] write(obj::Vector{Vector{Bool}})
   @ Avro ~/.julia/packages/Avro/JEoRa/src/types/binary.jl:22
 [4] top-level scope
   @ REPL[31]:1
```

Something appears to be going wrong with empty arrays (but only if they are nested inside of other arrays).

Julia 1.6.2, Avro 1.0.0.

--- a/src/types/arrays.jl
+++ b/src/types/arrays.jl
@@ -48,8 +48,12 @@ function writevalue(B::Binary, A::ArrayType, x, buf, pos, len, opts)
 end
 
 function nbytes(A::ArrayType, x, self=false)
-    nb = isempty(x) ? 0 : sum(y -> nbytes(A.items, y), x)
-    return self ? nb : nbytes(length(x)) + nbytes(nb) + nb + 1
+    if isempty(x)
+        return self ? 0 : 1
+    else
+        nb = sum(y -> nbytes(A.items, y), x)
+        return self ? nb : nbytes(length(x)) + nbytes(nb) + nb + 1
+    end
 end
 
 function readvalue(B::Binary, AT::ArrayType, ::Type{A}, buf, pos, buflen, opts) where {A <: AbstractVector{T}} where {T}

